tests/e2e/utils/helpers.py

```diff
@@ -0,0 +1,97 @@
+import json
+from datetime import datetime
+from functools import lru_cache
+from typing import Any, Optional, Union
+
+from pydantic import BaseModel
+from retry import retry
+
+
+# Helper methods && Class
+class Log(BaseModel):
+    level: str
+    location: str
+    message: Union[dict, str]
+    timestamp: str
+    service: str
+    cold_start: Optional[bool]
+    function_name: Optional[str]
+    function_memory_size: Optional[str]
+    function_arn: Optional[str]
+    function_request_id: Optional[str]
+    xray_trace_id: Optional[str]
+    extra_info: Optional[str]
+
+
+def trigger_lambda(lambda_arn: str, client: Any):
+    response = client.invoke(FunctionName=lambda_arn, InvocationType="RequestResponse")
+    return response
+
+
+@lru_cache(maxsize=10, typed=False)
+@retry(ValueError, delay=1, jitter=1, tries=10)
+def get_logs(lambda_function_name: str, log_client: Any, start_time: int, **kwargs: dict):
+    response = log_client.filter_log_events(logGroupName=f"/aws/lambda/{lambda_function_name}", startTime=start_time)
+    if not response["events"]:
+        raise ValueError("Empty response from Cloudwatch Logs. Repeating...")
+    filtered_logs = []
+    for event in response["events"]:
+        try:
+            message = Log(**json.loads(event["message"]))
+        except json.decoder.JSONDecodeError:
+            continue
+        filtered_logs.append(message)
+
+    return filtered_logs
+
+
+@lru_cache(maxsize=10, typed=False)
+@retry(ValueError, delay=1, jitter=1, tries=10)
+def get_metrics(
+    namespace: str, cw_client: Any, start_date: datetime, end_date: datetime, metric_name: str, service_name: str
+):
+    response = cw_client.get_metric_data(
+        MetricDataQueries=[
```

Comment:
I thought you were gonna try the new shiny Metric Insights SQL :D /s

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch-metrics-insights-querylanguage.html

Reply:
Wasn't aware of this! Replacing code to use SQL